src/Nethermind/Nethermind.Blockchain/BeaconBlockRoot/BeaconBlockRootHandler.cs

```diff
@@ -0,0 +1,38 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Nethermind.Core.Specs;
+using Nethermind.Core;
+using Nethermind.Evm.Precompiles.Stateful;
+using Nethermind.Int256;
+using Nethermind.State;
+using static Nethermind.Evm.Precompiles.Stateful.BeaconBlockRootPrecompile;
+
+
+namespace Nethermind.Consensus.BeaconBlockRoot;
+public class BeaconBlockRootHandler : IBeaconBlockRootHandler
+{
+    public void HandleBeaconBlockRoot(Block block, IReleaseSpec spec, IWorldState stateProvider)
+    {
+        if (!spec.IsBeaconBlockRootAvailable) return;
```

Comment:
better would be depending on the spec to create this check, use actual one or null instance instead of passing it everywhere into constructors, would simplify a lot.

Reply:
Summarizing:
Rename this to `PrecompileBeaconBlockRootHandler`
Have a `BeaconBlockRootHandler` that would call either this one or `NullHandler` depending on the spec. 
Don't pass it in but create it in block processor.

WDYT? (Just a proposition for now)